src/diffusers/pipelines/README.md

```diff
@@ -1,19 +1,168 @@
+- **Easy-to-tweak**: Certain pipelines will not be able to handle all use cases and tasks that you might like them to. If you want to use a certain pipeline for a specific use case that is not yet supported, you might have to copy the pipeline file and tweak the code to your needs.
+
+We try to make the pipeline code as readable as possible so that each part from pre-processing to diffusing to post-processing can easily be adapted. If you would like the community to benefit from your customized pipeline, we would ❤️  to see a contribution to our [community-examples](https://github.com/huggingface/diffusers/tree/main/examples/commmunity). If however you feel like an important pipeline is missing that deserves to be among the official pipelines in your opinion, a contribution to the [official pipeplines](https://github.com/huggingface/diffusers/blob/main/src/diffusers/pipelines) would be even better 🤗.
+- **One-purpose-only**: Pipelines should be used for one task and one task only. Even two tasks are from a modeling point of view very similar, *e.g.* image2image translation and in painting, pipelines shall be used for one task only to keep them *easy-to-tweak* and *readable*.
 
 ## Examples
 
-- DDPM for unconditional image generation in [pipeline_ddpm](https://github.com/huggingface/diffusers/blob/main/src/diffusers/pipelines/ddpm/pipeline_ddpm.py).
-- DDIM for unconditional image generation in [pipeline_ddim](https://github.com/huggingface/diffusers/blob/main/src/diffusers/pipelines/ddim/pipeline_ddim.py).
-- PNDM for unconditional image generation in [pipeline_pndm](https://github.com/huggingface/diffusers/blob/main/src/diffusers/pipelines/pndm/pipeline_pndm.py).
-- Latent diffusion for text to image generation / conditional image generation in [pipeline_latent_diffusion](https://github.com/huggingface/diffusers/blob/main/src/diffusers/pipelines/latent_diffusion/pipeline_latent_diffusion.py).
-- Glide for text to image generation / conditional image generation in [pipeline_glide](https://github.com/huggingface/diffusers/blob/main/src/diffusers/pipelines/glide/pipeline_glide.py).
-- BDDMPipeline for spectrogram-to-sound vocoding in [pipeline_bddm](https://github.com/huggingface/diffusers/blob/main/src/diffusers/pipelines/bddm/pipeline_bddm.py).
-- Grad-TTS for text to audio generation / conditional audio generation in [pipeline_grad_tts](https://github.com/huggingface/diffusers/blob/main/src/diffusers/pipelines/grad_tts/pipeline_grad_tts.py).
+### Image-to-Image text-guided generation with Stable Diffusion
+
+The `image_to_image.py` script implements `StableDiffusionImg2ImgPipeline`. It lets you pass a text prompt and an initial image to condition the generation of new images. This example also showcases how you can write custom diffusion pipelines using `diffusers`!
+
+```python
+from torch import autocast
+import requests
+from PIL import Image
+from io import BytesIO
+
+from image_to_image import StableDiffusionImg2ImgPipeline, preprocess
```

Comment:
Shouldn't this be `from diffusers.pipeline import StableDiffusionInPaintPipeline`? Perhaps we should also rename `preprocess` to something like `preprocess_for_img2img_input` (we need to come up with something better), and export the symbol in at least `pipelines/stable_diffusion/__init__.py`.

In addition, shouldn't this example become the contents of `examples/inference/image_to_image.py`? Currently the file is empty except for a single line that imports the pipeline.

Reply:
`examples/inference` should be fully deprecated - also ideally the preprocess function is not called outside of the pipeline but only inside the pipeline